### PR TITLE
Cleanups to resident runner and flutter device classes

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Remove some cruft that has built up over the years.

- Remove FlutterDevice supporting multiple VMServices
- Remove some excessive usage of Completers in reload/restart.

### CODE REVIEW

File: packages/flutter_tools/lib/src/run_hot.dart [hunk 1]
```diff
@@ -851,7 +842,7 @@ class HotRunner extends ResidentRunner {
           }
         }(),
     ];
-    final Future<void> reassembleFuture = Future.wait<void>(futures).then<void>((List<void> values) { });
+    final Future<void> reassembleFuture = Future.wait<void>(futures);
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/run_hot.dart]: Not sure what this was doing.

File: packages/flutter_tools/lib/src/ios/devices.dart [hunk 2]
```diff
@@ -578,33 +578,30 @@ class IOSDeviceLogReader extends DeviceLogReader {
...
 
   @override
-  List<VMService> get connectedVMServices => _connectedVMServices;
-  List<VMService> _connectedVMServices;
+  VMService get connectedVMService => _connectedVMService;
+  VMService _connectedVMService;
 
   @override
-  set connectedVMServices(List<VMService> connectedVMServices) {
+  set connectedVMService(VMService connectedVMServices) {
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/ios/devices.dart]: nit. the parameter should be `connectedVmService`

File: packages/flutter_tools/lib/src/resident_runner.dart [hunk 3]
```diff
@@ -208,36 +208,24 @@ class FlutterDevice {
...
-    final List<Future<void>> futures = <Future<void>>[
-      for (VMService service in vmServices) service.vm.refreshViews(waitForViews: true),
-    ];
-    await Future.wait(futures);
+    return vmService.vm.refreshViews(waitForViews: true);
   }
 
   List<FlutterView> get views {
-    if (vmServices == null) {
+    if (vmService == null || vmService.isClosed) {
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/resident_runner.dart]: nice

File: packages/flutter_tools/lib/src/resident_runner.dart [hunk 4]
```diff
@@ -208,36 +208,24 @@ class FlutterDevice {
...
+    return (viewFilter != null
+        ? vmService.vm.allViewsWithName(viewFilter)
+        : vmService.vm.views).toList();
   }
 
   Future<void> getVMs() async {
-    for (VMService service in vmServices) {
-      await service.getVM();
-    }
+    return vmService.getVM();
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/resident_runner.dart]: does this not have to be `await`ed?

File: packages/flutter_tools/lib/src/run_hot.dart (hunk 1 above)
Comment (christopherfujino) [reply packages/flutter_tools/lib/src/run_hot.dart]: hmm...

File: packages/flutter_tools/lib/src/resident_runner.dart (hunk 4 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/resident_runner.dart]: returning a future is approximately the same as awaiting a future and then returning

Comment (zanderso) [reply packages/flutter_tools/lib/src/resident_runner.dart]: Maybe:
```dart
Future<void> getVMs() => vmService.getVM();
```

**Review by zanderso — APPROVED** [review]

File: packages/flutter_tools/lib/src/ios/devices.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/ios/devices.dart]: Fixed!

File: packages/flutter_tools/lib/src/resident_runner.dart (hunk 4 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/resident_runner.dart]: Done

**Review by christopherfujino — APPROVED** [review]
lgtm

**Pull request merged**